v0/destinations/mp/deleteUsers.js

```diff
@@ -20,33 +23,30 @@ const userDeletionHandler = async (userAttributes, config) => {
       .setStatus(400)
       .build();
   }
-  const params = {
-    data: {
-      $token: `${config.token}`,
-      $delete: null,
-      $ignore_alias: true
+  const endpoint = "https://api.mixpanel.com/engage#profile-delete";
+  const data = [];
+  userAttributes.forEach(userAttribute => {
+    // Dropping the user if userId is not present
+    if (userAttribute.userId) {
+      data.push({
+        $token: `${config.token}`,
+        $distinct_id: userAttribute.userId,
+        $delete: null,
+        $ignore_alias: true
```

Comment:
Apart from userId every attribute is the same for all. We can store them in a variable and use, WDYT.